applications/GeoMechanicsApplication/python_scripts/geomechanics_solvers_wrapper.py

```diff
@@ -1,16 +1,21 @@
 import KratosMultiphysics
 from importlib import import_module
 
-def CreateSolver(model, custom_settings):
 
+def CreateSolverByParameters(model, custom_settings, parallelism):
     if (type(model) != KratosMultiphysics.Model):
         raise Exception("input is expected to be provided as a Kratos Model object")
 
     if (type(custom_settings) != KratosMultiphysics.Parameters):
         raise Exception("input is expected to be provided as a Kratos Parameters object")
 
-    parallelism = custom_settings["problem_data"]["parallel_type"].GetString()
-    solver_type_raw = custom_settings["solver_settings"]["solver_type"].GetString()
+    if custom_settings.Has("solver_settings"):
+        params = custom_settings["solver_settings"]
+        #To conform with other apps where only the "solver_settings" are passed
+    else:
+        params = custom_settings
```

Comment:
As mentioned in the comment for line 44, we believe we don't need to check if we're dealing with 'general' parameters or 'solver_settings' if we pass on the solver_settings from the CreateSolver function